keps/sig-cluster-lifecycle/0015-kubeadm-join-master.md

```diff
@@ -0,0 +1,431 @@
+reviewers:
+  - "@cha”
+  - "@jdtibier"
+approvers:
+  - "@luxas"
+  - "@timothysc"
+editor:
+  - "@fabriziopandini"
+creation-date: 2018-01-28
+last-updated: 2018-06-29
+see-also:
+  - KEP 0004
+```
+
+## Table of Contents
+
+<!-- TOC -->
+
+- [kubeadm join --master workflow](#kubeadm-join---master-workflow)
+    - [Metadata](#metadata)
+    - [Table of Contents](#table-of-contents)
+    - [Summary](#summary)
+    - [Motivation](#motivation)
+        - [Goals](#goals)
+        - [Non-goals](#non-goals)
+        - [Challenges and Open Questions](#challenges-and-open-questions)
+    - [Proposal](#proposal)
+        - [User Stories](#user-stories)
+            - [Create a cluster with more than one master nodes (static workflow)](#create-a-cluster-with-more-than-one-master-nodes-static-workflow)
+            - [Add a new master node (dynamic workflow)](#add-a-new-master-node-dynamic-workflow)
+        - [Implementation Details](#implementation-details)
+            - [Initialize the Kubernetes cluster](#initialize-the-kubernetes-cluster)
+            - [Preparing for execution of kubeadm join --master](#preparing-for-execution-of-kubeadm-join---master)
+            - [The kubeadm join --master workflow](#the-kubeadm-join---master-workflow)
+            - [dynamic workflow (advertise-address == `controlplaneAddress`)](#dynamic-workflow-advertise-address--controlplaneaddress)
+            - [Static workflow (advertise-address != `controlplaneAddress`)](#static-workflow-advertise-address--controlplaneaddress)
+            - [Strategies for deploying control plane components](#strategies-for-deploying-control-plane-components)
+            - [Strategies for distributing cluster certificates](#strategies-for-distributing-cluster-certificates)
+            - [`kubeadm upgrade` for HA clusters](#kubeadm-upgrade-for-ha-clusters)
+    - [Graduation Criteria](#graduation-criteria)
+    - [Implementation History](#implementation-history)
+    - [Drawbacks](#drawbacks)
+    - [Alternatives](#alternatives)
+
+<!-- /TOC -->
+
+## Summary
+
+We are extending the kubeadm distinctive `init` and `join` workflow, introducing the
+capability to add more than one master node to an existing cluster by means of the
+new `kubeadm join --master` option (in alpha release the flag will be named --experimental-master)
+
+As a consequence, kubeadm will provide a best-practice, “fast path” for creating a
+minimum viable, conformant Kubernetes cluster with one or more master nodes and
+zero or more worker nodes.
+
+## Motivation
+
+Support for high availability is one of the most requested features for kubeadm.
+
+Even if, as of today, there is already the possibility to create an HA cluster
+using kubeadm in combination with some scripts and/or automation tools (e.g.
+[this](https://kubernetes.io/docs/setup/independent/high-availability/)), this KEP was
+designed with the objective to introduce an upstream simple and reliable solution for
+achieving the same goal.
```

Comment:
... with important non-goals. Say clearly (also in the summary) that this doesn't solve every case or even the full end-to-end flow automatically

Reply:
done